src/diffusers/models/transformer.py

```diff
@@ -0,0 +1,189 @@
+import math
+from dataclasses import dataclass
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..configuration_utils import ConfigMixin, register_to_config
+from ..modeling_utils import ModelMixin
+from ..utils import BaseOutput
+
+
+@dataclass
+class MaskedLMOutput(BaseOutput):
+    """
+    Base class for masked language models outputs.
+
+    Args:
+        logits (`torch.FloatTensor` of shape `(batch_size, sequence_length, config.vocab_size)`):
+            Prediction scores of the language modeling head (scores for each vocabulary token before SoftMax).
+    """
+
+    logits: torch.FloatTensor = None
```

Comment:
Question; should we implement a fully-fledged Transformer like in transformers, which also outputs a list of hidden states and attention scores?